framework/wazuh/core/tests/test_wlogging.py

```diff
@@ -0,0 +1,159 @@
+@patch.object(wlogging.WazuhLogger, 'tag', create=True, new_callable=PropertyMock)
+@patch.object(wlogging.WazuhLogger, 'logger', create=True, new_callable=PropertyMock)
+@patch.object(wlogging.WazuhLogger, 'foreground_mode', create=True, new_callable=PropertyMock)
+@patch.object(wlogging.WazuhLogger, 'debug_level', create=True, new_callable=PropertyMock)
+@patch.object(wlogging.WazuhLogger, 'logger_name', create=True, new_callable=PropertyMock)
+@patch.object(wlogging.WazuhLogger, 'custom_formatter', create=True, new_callable=PropertyMock)
+def test_wazuh_logger__init__(mock_formatter, mock_logger_name, mock_debug_level, mock_foreground_mode,
+                              mock_logger, mock_tag, mock_log_path):
+    """Test if WazuhLogger __init__ method initialize all attributes properly.
+
+    Parameters
+    ----------
+    mock_formatter: PropertyMock
+        Mock custom_formatter attribute.
+    mock_logger_name: PropertyMock
+        Mock name for the logger.
+    mock_debug_level: PropertyMock
+        Mock log level.
+    mock_foreground_mode: PropertyMock
+        Mock foreground mode.
+    mock_logger: PropertyMock
+        Mock logger attribute.
+    mock_tag: PropertyMock
+        Mock tag attribute.
+    mock_log_path: Property
+        Mock path for the log.
+    """
+    # To bypass the checking of the existence of a valid Wazuh install
+    with patch('os.path.join'):
+        test = wlogging.WazuhLogger(foreground_mode=mock_foreground_mode, log_path=mock_log_path, tag=mock_tag,
+                                    debug_level=mock_debug_level, logger_name=mock_logger_name,
+                                    custom_formatter=mock_formatter)
+    for x in [mock_formatter, mock_logger_name, mock_debug_level, mock_foreground_mode,
+              mock_logger, mock_tag, mock_log_path]:
+        x.assert_called()
+
+
+@pytest.mark.parametrize('attribute, expected_exception', [
+    ('level', None),
+    ('foreground_mode', None),
+    ('doesnt_exists', AttributeError)
+])
+@patch('wazuh.core.wlogging.CustomFileRotatingHandler')
+def test_wazuh_logger_getattr(mock_fh, attribute, expected_exception):
+    """Test if WazuhLogger __getattr__ method works properly.
+
+    Parameters
+    ----------
+    mock_fh: MagicMock
+        Mock of CustomFileRotatingHandler function.
+    attribute: str
+        Attribute to search for with __getattr__.
+    expected_exception: Union[None, Exception]
+        Exception expected to be raised.
+    """
+    tmp_dir = tempfile.TemporaryDirectory
+    # To bypass the checking of the existence of a valid Wazuh install
+    with patch('os.path.join'):
+        w_logger = wlogging.WazuhLogger(foreground_mode=True, log_path=tmp_dir, tag='{test} {test}: {test}',
+                                        debug_level=[0, 'test'], logger_name='test')
+    w_logger.setup_logger()
+
+    if expected_exception is None:
+        w_logger.__getattr__(attribute)
```

Comment:
Perhaps an assertion here would be useful to check that the response is as expected.

Reply:
Added in 271cdf0cb8089c853adc3eb5ad26a6638aae9606